delfin/task_manager/tasks/telemetry.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+import six
+from oslo_log import log
+
+from delfin import context
+from delfin.drivers import api as driver_api
+from delfin.exporter import base_exporter
+
+LOG = log.getLogger(__name__)
+
+
+class TelemetryTask(object):
+    @abc.abstractmethod
+    def collect(self, ctx, storage_id, args, start_time, end_time):
+        pass
+
+
+class PerformanceCollectionTask(TelemetryTask):
+    def __init__(self):
+        self.driver_api = driver_api.API()
+        self.perf_exporter = base_exporter.PerformanceExporterManager()
+
+    def collect(self, ctx, storage_id, args, start_time, end_time):
+        try:
+            perf_metrics = self.driver_api \
+                .collect_perf_metrics(ctx, storage_id,
+                                      args,
+                                      start_time, end_time)
+
+            self.perf_exporter.dispatch(context, perf_metrics)
```

Comment:
Need to add some defuaut labels  before pushing. this code snippet was removed while removing old scheduler code.. 
# fill extra labels to metric by fetching metadata from resource DB
            try:
                array_details = db.storage_get(self.context, storage_id=self
                                               .storage_id)
                for m in array_metrics:
                    m.labels["name"] = array_details.name
                    m.labels["serial_number"] = array_details.serial_number

            except Exception as e:
                msg = _('Failed to add extra labels to array performance '
                        'metrics: {0}'.format(e))
                LOG.error(msg)

Reply:
Updated